File: app/src/main/java/com/battlelancer/seriesguide/movies/MoviesNowAdapter.java
package com.battlelancer.seriesguide.movies;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.battlelancer.seriesguide.shows.history.HistoryViewHolder;
import com.battlelancer.seriesguide.shows.history.NowAdapter;

/**
 * An adapted version of {@link NowAdapter} with a special layout for movies.
 */
class MoviesNowAdapter extends NowAdapter {

    MoviesNowAdapter(Context context, ItemClickListener listener) {
        super(context, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
        if (viewHolder instanceof HistoryViewHolder) {
            NowItem item = getItem(position);
            HistoryViewHolder holder = (HistoryViewHolder) viewHolder;
            holder.bindToMovie(getContext(), item, getDrawableWatched(), getDrawableCheckin());
        } else {
            super.onBindViewHolder(viewHolder, position);
        }
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/movies/MoviesNowAdapter.kt
// Copyright 2023 Uwe Trottmann
// SPDX-License-Identifier: Apache-2.0

package com.battlelancer.seriesguide.movies

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.battlelancer.seriesguide.shows.history.HistoryViewHolder
import com.battlelancer.seriesguide.shows.history.NowAdapter

/**
 * An adapted version of [NowAdapter] with a special layout for movies.
 */
internal class MoviesNowAdapter(context: Context, listener: ItemClickListener) :
    NowAdapter(context, listener) {

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        if (viewHolder is HistoryViewHolder) {
            val item = getItem(position)
            viewHolder.bindToMovie(context, item, drawableWatched, drawableCheckin)
        } else {
            super.onBindViewHolder(viewHolder, position)
        }
    }
}